evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -103,9 +104,11 @@ def _encode_labels(self, y):
         return y_encoded
 
     def fit(self, X, y=None):
+        X_encoded = _convert_to_woodwork_structure(X)
         X_encoded = self._encode_categories(X, fit=True)
         y_encoded = self._encode_labels(y)
-        return super().fit(X_encoded, y_encoded)
+        self._component_obj.fit(X_encoded, y_encoded)
```

Comment:
Updating to just call here, because Estimator will call `self.input_feature_names = list(X.columns)` but since we've changed the feature names in `_encode_categories` (since LGBM isn't compatible with weird names), this would override the original names.